class.photon.php

```diff
@@ -422,9 +421,30 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 			 * 	 @type $size Image size. Can be a string (name of the image size, e.g. full) or an integer.
 			 * }
 			 */
-			apply_filters( 'jetpack_photon_override_image_downsize', false, compact( 'image', 'attachment_id', 'size' ) )
-		)
+			false === apply_filters( 'jetpack_photon_admin_allow_image_downsize', false, compact( 'image', 'attachment_id', 'size' ) )
```

Comment:
You're right.

The use case would be for a site that does not generate any intermediate sizes and would want to rely on Photon fully (VIP Go, in this case). Do you think this is acceptable, if we add a clear warning about the implications of this or the alternative is break out the logic for usage from the actual functionality. VIP Go, or other site owners, could remove the default filter on `image_downsize` from Photon and add their own that directly calls a new function that performs the work without checking any logic?